src/sqlfluff/rules/L067.py

```diff
@@ -0,0 +1,156 @@
+"""Implementation of Rule L067."""
+from typing import Optional
+
+from sqlfluff.core.parser import NewlineSegment
+
+from sqlfluff.core.rules.base import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
+from sqlfluff.core.rules.doc_decorators import document_fix_compatible, document_groups
+from sqlfluff.utils.functional import sp, FunctionalContext
+
+
+@document_groups
+@document_fix_compatible
+class Rule_L067(BaseRule):
+    """A subquery inside parentheses should start on a new line.
+
+    For brackets, if there are any subqueries inside, the subqueries
+    should begin on the next line after the starting bracket, and the
+    ending bracket should go on the next line after the subquery.
+
+    **Anti-pattern**
+
+    Parentheses on the line with a subquery.
+
+    .. code-block:: sql
+        :force:
+
+        SELECT A.SOMETHING
+        FROM (SELECT
+                B,
+                C
+            FROM D GROUP BY 1) AS A
+
+    **Best practice**
+
+    Opening bracket and ending bracket take a new line out of the subquery.
+
+    .. code-block:: sql
+        :force:
+
+        SELECT A.SOMETHING
+        FROM (
+            SELECT
+                B,
+                C
+            FROM D GROUP BY 1
+        ) AS A
+
+    """
+
+    groups = ("all",)
+    crawl_behaviour = SegmentSeekerCrawler({"bracketed"})
+
+    def _eval(self, context: RuleContext) -> Optional[LintResult]:
+        # only care about bracket
+        assert context.segment.is_type("bracketed")
+
+        expressions = FunctionalContext(context).segment.children()
+        start_bracket = expressions.select(sp.is_type("start_bracket")).first().get()
+        end_bracket = (
+            expressions.reversed().select(sp.is_type("end_bracket")).first().get()
+        )
+
+        # Check whether there is any subquery found inside the brackets
+        select_targets = expressions.select(sp.is_type("select_statement"))
+        if select_targets:
+            results = []
+            # check if the select statement only take a single line.
+            # If so, the parentheses can keep at the same line with them.
+            if (
+                len(select_targets) == 1
+                and select_targets[0].get_start_loc()[0]
+                == select_targets[0].get_end_loc()[0]
+                == start_bracket.pos_marker.line_no
+                == end_bracket.pos_marker.line_no
```

Comment:
I don't think line number information is correct once other rule fixes have been applied (i.e. partway through linting). Can you look at other rules (such as L003 and L016) to see how they handle this?